contracts/finality/src/exec/finality.rs

```diff
@@ -38,13 +44,14 @@ pub fn handle_public_randomness_commit(
     // note that `insert_pub_rand_commit` has ensured that
     // - the new commitment does not overlap with the existing ones
     // - the new commitment does not have num_pub_rand = 0
+    let current_epoch = get_current_epoch(&deps.as_ref())?;
     insert_pub_rand_commit(
         deps.storage,
         fp_pubkey_hex,
         PubRandCommit {
             start_height,
             num_pub_rand,
-            height: env.block.height,
+            epoch: current_epoch,
```

Comment:
```suggestion
            babylon_genesis_epoch: current_epoch,
```

This might be a helpful indicator to the fact that this is the Babylon Genesis epoch.